src/components/Filters/index.tsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+import { Material } from '@balena/ui-shared-components';
+import { useTranslation } from '../../hooks/useTranslation';
+import { Search } from './Search';
+import { Views } from './Views';
+import { FiltersDialog } from './FiltersDialog';
+import {
+	type FormData,
+	createFullTextSearchFilter,
+	randomUUID,
+	FULL_TEXT_SLUG,
+	parseFilterDescription,
+} from './SchemaSieve';
+import { Summary } from './Summary';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faFilter } from '@fortawesome/free-solid-svg-icons/faFilter';
+import { useClickOutside } from '../../hooks/useClickOutside';
+import type { JSONSchema7 as JSONSchema } from 'json-schema';
+import { isJSONSchema } from '../../AutoUI/models/helpers';
+
+const { Box, Button, useTheme, useMediaQuery } = Material;
+
+export interface FiltersView {
+	id: string;
+	eventName: string;
+	name: string;
+	filters: JSONSchema[];
+}
+
+type FilterRenderMode = 'all' | 'add' | 'search' | 'views' | 'summary';
+
+export interface FiltersProps {
+	schema: JSONSchema;
+	dark?: boolean;
+	filters?: JSONSchema[];
+	views?: FiltersView[];
+	onFiltersChange?: (filters: JSONSchema[]) => void;
+	onViewsChange?: (views: FiltersView[]) => void;
+	renderMode?: FilterRenderMode | FilterRenderMode[];
```

Comment:
Not to be addressed during this PR, but I'm not quite happy with this property.

`AssignDialog` uses `search` and `summary`, however the user can still click the `summary` filters to change them into basically the same thing `add` does. Which means all we are excluding is `views`. Then `DeviceLogs` has all 4. Meanwhile, AutoUI has *one* Filter for `add`, `search`, and `views`, and then *another* Filter for `summary`.

So effectively what we amount to is we use `all` for all instances of Filters except the `AssignDialog`, for which we want to exclude views and the ability to add non-full-text-search filters (bugged atm), and for AutoUI, the main owner of filters, uses them in pieces instead of as a whole.

We should revisit this later for two things:
- The bug we see in AssignDialog where we hide `add` but can still access it via `summary`
- Analyzing whether we can reduce this to a single usecase

It's possible that whatever we choose to replace AutoUI will take care of this 🤞

Reply:
I agree, I also would like to remove some of the properties, for sure we can rethink renderMode, AutoUI has a limitation which is, having the summary in the same FIlters components make it really ugly to handle unfortunately. While doing it I also thought that we might soon switch to something different like a sidebar instead of having this filters view. If we rethink the graphics and the way we display them, we can even just remove some of these properties. It would be very nice